<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ProjectMenuPanel</class>
 <widget class="QWidget" name="ProjectMenuPanel">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>238</width>
    <height>822</height>
   </rect>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget {
    background-color: #212529;
}
QLabel#project_name {
    font-size: 14px; 
    font-weight: bold; 
    color: white; 
    padding-bottom: 0px;
}
QFrame#desc_box {
    background-color: #495057;
    border-radius: 6px;
    border: 1px solid #adb5bd;
    margin-top: 8px;
    min-height: 90px;
    max-height: 150px;
}
QTextEdit#desc_text {
    color: #e9ecef;
    background: transparent;
    border: none;
    padding: 0px;
    margin: 0px;
    min-height: 90px;
    max-height: 150px;
}
QFrame#data_section, QFrame#flow_section {
    background-color: #495057;
    border-radius: 6px;
    border: 1px solid #adb5bd;
    margin-top: 8px;
    height: 100px;
}
QLabel#data_label, QLabel#flow_label {
    font-size: 14px; 
    font-weight: bold; 
    color: white; 
    padding: 0px;
    margin: 0px;
}
QPushButton#data_add_btn, QPushButton#flow_add_btn {
    border: none;
    background-color: transparent;
    padding: 0px;
    margin: 0px;
    margin-top: 2px;
}
QFrame#data_list_container, QFrame#flow_list_container {
    background-color: #343a40;
    border: 1px solid #adb5bd;
    border-radius: 4px;
    min-height: 120px;
}
QLabel#no_data_label, QLabel#no_flow_label {
    color: #adb5bd;
    font-style: italic;
    font-size: 12px;
}
QPushButton#close_btn {
    background-color: #6c757d;
    padding: 8px;
    border-radius: 6px;
    text-align: center;
    font-size: 16px;
    font-weight: bold;
    border: 1px solid #adb5bd;
    color: white;
    min-height: 32px;
}
QPushButton#close_btn:hover {
    background-color: #868e96;
    border-color: #ced4da;
}
QScrollArea {
    border: none;
    background: transparent;
    border-radius: 6px;
    padding: 0px;
    margin: 0px;
}
QScrollArea QWidget#desc_container, QScrollArea QWidget#data_list_container {
    background: transparent;
    border-radius: 6px;
    padding: 0px;
    margin: 0px;
}
QScrollArea QFrame {
    border: 1px solid #adb5bd;
    border-radius: 6px;
    background-color: #495057;
    padding: 0px;
    margin: 0px;
}
QScrollBar:vertical {
    border: none;
    background: #495057;
    width: 8px;
    margin: 0px;
    position: absolute;
    right: 0px;
}
QScrollBar::handle:vertical {
    background: #adb5bd;
    border-radius: 4px;
    min-height: 20px;
}
QScrollBar::handle:vertical:hover {
    background: #ced4da;
}
QScrollBar::add-line:vertical, QScrollBar::sub-line:vertical {
    height: 0px;
}
QScrollBar::add-page:vertical, QScrollBar::sub-page:vertical {
    background: transparent;
}
</string>
  </property>
  <layout class="QVBoxLayout" name="menu_layout">
   <property name="spacing">
    <number>8</number>
   </property>
   <property name="leftMargin">
    <number>8</number>
   </property>
   <property name="topMargin">
    <number>20</number>
   </property>
   <property name="rightMargin">
    <number>8</number>
   </property>
   <property name="bottomMargin">
    <number>16</number>
   </property>
   <item>
    <widget class="QFrame" name="project_section">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::Panel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Shadow::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>0</number>
     </property>
     <layout class="QVBoxLayout" name="project_name_layout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>4</number>
      </property>
      <property name="rightMargin">
       <number>4</number>
      </property>
      <property name="bottomMargin">
       <number>4</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="project_name_header">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>4</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="project_name_label_layout">
          <property name="spacing">
           <number>5</number>
          </property>
          <property name="leftMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QLabel" name="project_name">
            <property name="font">
             <font>
              <pointsize>-1</pointsize>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>프로젝트 이름</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="open_folder_btn">
            <property name="mouseTracking">
             <bool>false</bool>
            </property>
            <property name="styleSheet">
             <string notr="true">border:0</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="iconSize">
             <size>
              <width>14</width>
              <height>14</height>
             </size>
            </property>
            <property name="fixedSize" stdset="0">
             <size>
              <width>18</width>
              <height>18</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QFrame" name="project_box">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>90</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>150</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="frameShape">
         <enum>QFrame::Shape::Panel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Shadow::Plain</enum>
        </property>
        <layout class="QVBoxLayout" name="desc_layout">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>4</number>
         </property>
         <property name="topMargin">
          <number>4</number>
         </property>
         <property name="rightMargin">
          <number>4</number>
         </property>
         <property name="bottomMargin">
          <number>12</number>
         </property>
         <item>
          <widget class="QScrollArea" name="desc_scroll">
           <property name="frameShape">
            <enum>QFrame::Shape::NoFrame</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Shadow::Plain</enum>
           </property>
           <property name="verticalScrollBarPolicy">
            <enum>Qt::ScrollBarPolicy::ScrollBarAsNeeded</enum>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
           </property>
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="desc_container">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>208</width>
              <height>132</height>
             </rect>
            </property>
            <property name="styleSheet">
             <string notr="true">background: transparent;</string>
            </property>
            <layout class="QVBoxLayout" name="desc_container_layout">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="leftMargin">
              <number>4</number>
             </property>
             <property name="topMargin">
              <number>4</number>
             </property>
             <property name="rightMargin">
              <number>4</number>
             </property>
             <property name="bottomMargin">
              <number>4</number>
             </property>
             <item>
              <widget class="QTextEdit" name="desc_text">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>90</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>150</height>
                </size>
               </property>
               <property name="contextMenuPolicy">
                <enum>Qt::ContextMenuPolicy::CustomContextMenu</enum>
               </property>
               <property name="styleSheet">
                <string notr="true">
                 QTextEdit {
                     color: #e9ecef;
                     background: transparent;
                     border: none;
                     padding: 0px;
                     margin: 0px;
                 }
                 QScrollBar:vertical {
                     border: none;
                     background: #495057;
                     width: 8px;
                     margin: 0px;
                     position: absolute;
                     right: 0px;
                 }
                 QScrollBar::handle:vertical {
                     background: #adb5bd;
                     border-radius: 4px;
                     min-height: 20px;
                 }
                 QScrollBar::handle:vertical:hover {
                     background: #ced4da;
                 }
                 QScrollBar::add-line:vertical, QScrollBar::sub-line:vertical {
                     height: 0px;
                 }
                 QScrollBar::add-page:vertical, QScrollBar::sub-page:vertical {
                     background: transparent;
                 }
                </string>
               </property>
               <property name="verticalScrollBarPolicy">
                <enum>Qt::ScrollBarPolicy::ScrollBarAsNeeded</enum>
               </property>
               <property name="horizontalScrollBarPolicy">
                <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
               </property>
               <property name="placeholderText">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="data_section">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::Panel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Shadow::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>0</number>
     </property>
     <layout class="QVBoxLayout" name="data_layout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>4</number>
      </property>
      <property name="rightMargin">
       <number>4</number>
      </property>
      <property name="bottomMargin">
       <number>4</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="data_header">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>4</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="label_layout">
          <property name="spacing">
           <number>5</number>
          </property>
          <property name="leftMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QLabel" name="data_label">
            <property name="font">
             <font>
              <pointsize>-1</pointsize>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>데이터 소스</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="data_add_btn">
            <property name="iconSize">
             <size>
              <width>14</width>
              <height>14</height>
             </size>
            </property>
            <property name="fixedSize" stdset="0">
             <size>
              <width>18</width>
              <height>18</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QScrollArea" name="data_scroll">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>100</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>200</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::Shape::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Shadow::Plain</enum>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarPolicy::ScrollBarAsNeeded</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QFrame" name="data_list_container">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>216</width>
           <height>200</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>100</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::Panel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Plain</enum>
         </property>
         <layout class="QVBoxLayout" name="data_list_layout">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="leftMargin">
           <number>8</number>
          </property>
          <property name="topMargin">
           <number>8</number>
          </property>
          <property name="rightMargin">
           <number>8</number>
          </property>
          <property name="bottomMargin">
           <number>8</number>
          </property>
          <item>
           <widget class="QListWidget" name="data_list">
            <property name="styleSheet">
             <string notr="true">
              QListWidget {
                  background: transparent;
                  border: none;
                  color: #e9ecef;
              }
              QListWidget::item {
                  padding: 4px;
                  border-radius: 4px;
              }
              QListWidget::item:selected, QListWidget::item:hover {
                  background: #6c757d;
              }
             </string>
            </property>
            <property name="frameShape">
             <enum>QFrame::Shape::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Shadow::Plain</enum>
            </property>
            <property name="verticalScrollBarPolicy">
             <enum>Qt::ScrollBarPolicy::ScrollBarAsNeeded</enum>
            </property>
            <property name="horizontalScrollBarPolicy">
             <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="flow_section">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::Panel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Shadow::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>0</number>
     </property>
     <layout class="QVBoxLayout" name="flow_layout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>4</number>
      </property>
      <property name="rightMargin">
       <number>4</number>
      </property>
      <property name="bottomMargin">
       <number>4</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="flow_header">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>4</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="flow_label_layout">
          <property name="spacing">
           <number>5</number>
          </property>
          <property name="leftMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QLabel" name="flow_label">
            <property name="font">
             <font>
              <pointsize>-1</pointsize>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>흐름</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="flow_add_btn">
            <property name="fixedSize" stdset="0">
             <size>
              <width>18</width>
              <height>18</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QScrollArea" name="flow_scroll">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>100</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>200</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::Shape::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Shadow::Plain</enum>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarPolicy::ScrollBarAsNeeded</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QFrame" name="flow_list_container">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>216</width>
           <height>100</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>100</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::Panel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Plain</enum>
         </property>
         <layout class="QVBoxLayout" name="flow_list_layout">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="leftMargin">
           <number>8</number>
          </property>
          <property name="topMargin">
           <number>8</number>
          </property>
          <property name="rightMargin">
           <number>8</number>
          </property>
          <property name="bottomMargin">
           <number>8</number>
          </property>
          <item>
           <widget class="QListWidget" name="flow_list">
            <property name="styleSheet">
             <string notr="true">
              QListWidget {
                  background: transparent;
                  border: none;
                  color: #e9ecef;
              }
              QListWidget::item {
                  padding: 4px;
                  border-radius: 4px;
              }
              QListWidget::item:selected, QListWidget::item:hover {
                  background: #6c757d;
              }
             </string>
            </property>
            <property name="frameShape">
             <enum>QFrame::Shape::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Shadow::Plain</enum>
            </property>
            <property name="horizontalScrollBarPolicy">
             <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QPushButton" name="close_btn">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>50</height>
      </size>
     </property>
     <property name="text">
      <string>프로젝트 닫기</string>
     </property>
     <property name="iconSize">
      <size>
       <width>16</width>
       <height>16</height>
      </size>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="resources/s-dia.qrc"/>
 </resources>
 <connections/>
</ui>
